### ISSUE TITLE
"triton profiles" should not require a current profile

### ISSUE THREAD

**Issue by jclulow** [joyent/node-triton#86]
Using `triton 4.4.3`, I tried running `triton profiles`:

```
$ triton profiles
triton profiles: error: opts.profile (object) is required
```

This is not extremely helpful.  I believe it's because I don't have `TRITON_PROFILE` set in my environment, and I didn't pass a `-p` flag.  But I _do_ have profiles in `$HOME/.triton/profiles.d`.

It'd be good to at least print an error that describes what I should do next, though it seems better to actually print the list of profiles and warn the user that one is not selected.

**Comment by jclulow** [comment]
Note, additionally, that I can't use `triton profile set-current` without a profile:

```
$ triton profile set-current coal
triton profile: error: opts.profile (object) is required
```

**Comment by trentm** [comment]
Josh didn't have '~/.triton/config.json' at the time. `triton` should cope.

**Comment by trentm** [comment]
Repro requires no config.json (at least no 'profile' value in it) *plus* no `TRITON_` or `SDC_` envvars. The latter because lacking a configured current profile, `triton` defaults to the virtual 'env' profile which uses the envvars.

```
$ set | grep TRITON_

$ set | grep SDC_

$ ls ~/.triton/config.json
ls: /Users/trentm/.triton/config.json: No such file or directory

$ triton -v profiles 2>&1 | bunyan
[2016-02-03T05:02:10.236Z] TRACE: triton/36272 on danger0.local (/Users/trentm/joy/node-triton/lib/cli.js:281): loaded config
    config: {
      "cacheDir": "cache",
      "_defaults": {
        "cacheDir": "cache"
      },
      "_configDir": "/Users/trentm/.triton"
    }
[2016-02-03T05:02:10.238Z] TRACE: triton/36272 on danger0.local (/Users/trentm/joy/node-triton/lib/cli.js:289): loaded profile (profile=null)
[2016-02-03T05:02:10.240Z] TRACE: triton/36272 on danger0.local (/Users/trentm/joy/node-triton/lib/cli.js:306 in fini): cli fini (subcmd=profiles)
    AssertionError: opts.profile (object) is required
        at new TritonApi (/Users/trentm/joy/node-triton/lib/tritonapi.js:76:12)
        at Object.createClient (/Users/trentm/joy/node-triton/lib/tritonapi.js:994:16)
        at CLI.tritonapi (/Users/trentm/joy/node-triton/lib/cli.js:291:41)
        at _listProfiles (/Users/trentm/joy/node-triton/lib/do_profile/do_list.js:43:33)
        at ProfileCLI.do_list (/Users/trentm/joy/node-triton/lib/do_profile/do_list.js:81:5)
        at ProfileCLI.dispatch (/Users/trentm/joy/node-triton/node_modules/cmdln/lib/cmdln.js:900:17)
        at CLI.do_profiles (/Users/trentm/joy/node-triton/lib/do_profiles.js:14:39)
        at CLI.dispatch (/Users/trentm/joy/node-triton/node_modules/cmdln/lib/cmdln.js:900:17)
        at /Users/trentm/joy/node-triton/node_modules/cmdln/lib/cmdln.js:425:18
        at CLI.init (/Users/trentm/joy/node-triton/node_modules/cmdln/lib/cmdln.js:638:5)
triton profiles: error: AssertionError: opts.profile (object) is required
    at new TritonApi (/Users/trentm/joy/node-triton/lib/tritonapi.js:76:12)
    at Object.createClient (/Users/trentm/joy/node-triton/lib/tritonapi.js:994:16)
    at CLI.tritonapi (/Users/trentm/joy/node-triton/lib/cli.js:291:41)
    at _listProfiles (/Users/trentm/joy/node-triton/lib/do_profile/do_list.js:43:33)
    at ProfileCLI.do_list (/Users/trentm/joy/node-triton/lib/do_profile/do_list.js:81:5)
    at ProfileCLI.dispatch (/Users/trentm/joy/node-triton/node_modules/cmdln/lib/cmdln.js:900:17)
    at CLI.do_profiles (/Users/trentm/joy/node-triton/lib/do_profiles.js:14:39)
    at CLI.dispatch (/Users/trentm/joy/node-triton/node_modules/cmdln/lib/cmdln.js:900:17)
    at /Users/trentm/joy/node-triton/node_modules/cmdln/lib/cmdln.js:425:18
    at CLI.init (/Users/trentm/joy/node-triton/node_modules/cmdln/lib/cmdln.js:638:5)
```

**Comment by trentm** [comment]
Fixed in commit edb28b6791d286a88c57d1435f8b2b7f4e621d91. Published in version 4.4.3.
